### DISCUSSION TITLE
Converting XML to ER7 and vice versa

### DISCUSSION THREAD

**Discussion by Caraja28** [nextgenhealthcare/connect#6011] [Q&A] [answered]
Greetings!
This is my first time posting something and i couldnt find anything similar anywhere.

Is there a way to tell Mirth to use the HL7 Data Types as Node Names instead(child objects only of course)?
Instead of:
<MSH.3>
            <**MSH.3.1**>SOME VALUE</MSH.3.1>
< /MSH.3>

I would like to get this:

<MSH.3>
            <**HD.1**>SOME VALUE</HD.1>
< /MSH.3>

I hoped there would be a simple way to switch this out as i do not intend to rename all of these myself.

Hopefully someone has already done this or there is a simple solution not involving a lookup table for every field....

**Comment by jonbartels** [comment, marked as answer]
Reference - https://hl7-definition.caristix.com/v2/HL7v2.5/Fields/MSH.3

A solution might be to use a different HL7 parser. Is there a Java-compatible HL7 parser that does what you want? I don't think Nule does. What about HAPI?

Does the strict parser in Mirth help you any?

Why do you want this format? What is your use case? 

Interesting problem.

**Comment by tonygermano** [comment]
As @jonbartels mentioned, the strict parser should take care of this. It uses HAPI under the hood and produces valid hl7v2 XML rather than the default mirth XML format.

https://docs.nextgen.com/bundle/Mirth_User_Guide_40/page/connect/connect/topics/c_HL7_v2_x_Data_Type_connect_ug.html

**Reply by Caraja28** [reply]
Thank you too!!!
